src/Mnemonic.cpp

```diff
@@ -26,8 +26,12 @@ inline const char* const* mnemonicWordlist() { return wordlist; }
 
 bool Mnemonic::isValidWord(const std::string& word) {
     const char* wordC = word.c_str();
+    const auto len = word.length();
     for (const char* const* w = mnemonicWordlist(); *w != nullptr; ++w) {
-        if (strncmp(*w, wordC, word.length()) == 0) {
+        if (strlen(*w) != len) {
+            continue;
+        }
+        if (strncmp(*w, wordC, len) == 0) {
```

Comment:
```suggestion
        if (strncmp(*w, wordC, strlen(*w)) == 0) {
```

Previous check makes sure that len == strlen(*w), but strlen(*w) is the correct here, that was the bug.  The length pre-check is not really needed.

Reply:
I just tried only `if (strncmp(*w, wordC, strlen(*w)) == 0)`, doesn't really work expected, got swift test errro: `Index out of range`,